### ISSUE TITLE
Support for input type having a list of complex input type ?

### ISSUE THREAD

**Issue by joel-costigliola** [Enigmatis/graphql-java-annotations#78]
Hi,

I would like to know if the annotations supports introspecting a `List` of complex input type.

Here's (simplified) example of what I'm trying to achieve which leads to a `Introspection must provide input type for arguments` error in chrome GraphiQL client .

I wonder if I'm doing something wrong or if there is no support of list of complex input types through annotations.

The `SpaceInstanceInput` below is what I'm passing as a parameter of this mutation

```java
@GraphQLField
public MutationResult updateSpaceInstance(
	  @GraphQLName("spaceInstanceInput") final SpaceInstanceInput spaceInstanceInput) {
```
where `SpaceInstanceInput` data structure looks like:

```java
public class SpaceInstanceInput {

    private final List<LogSubscriptionInput> logSubscriptions = null;
    private LogSubscriptionInput firstLogSubscriptions;

    // WORKS FINE
    @GraphQLField
    public LogSubscriptionInput getFirstLogSubscription() {
        return this.firstLogSubscriptions;
    }

    // DOES NOT WORK
    @GraphQLField
    public List<LogSubscriptionInput> getLogSubscriptions() {
        return this.logSubscriptions;
    }

}

public class LogSubscriptionInput {

    private LogTypeInput logTypeInput;
    private String subscriptionId;

    @GraphQLField
    public LogTypeInput getLogTypeInput() {
        return this.logTypeInput;
    }

    @GraphQLField
    public String getSubscriptionId() {
        return this.subscriptionId;
    }
}

public class LogTypeInput {

    private final String format;
    private final String type;

    @GraphQLField
    public String getFormat() {
        return this.format;
    }

    @GraphQLField
    public String getType() {
        return this.type;
    }
}
```

**Comment by joel-costigliola** [comment]
I have found that there is similar problem when using `@GraphQLNonNull` to wrap a complex input type like `LogTypeInput`.

**Comment by joel-costigliola** [comment]
I think the problem might come from [`geInputObject`](https://github.com/graphql-java/graphql-java-annotations/blob/cd18c2f7360ad28dba9460e4865195aa4b472b41/src/main/java/graphql/annotations/GraphQLAnnotations.java#L589) that should check whether it deals with an `GraphQLList` or `GraphQLNonNull` and look inside it (recursively) for object input types.

**Comment by mpires** [comment]
Thank you for that, I stumbled upon the same issue and that was helpful.

**Comment by apottere** [comment]
If someone wants to cook up a PR for this I can merge it, but I don't have time to look into it myself ⏳

**Comment by joel-costigliola** [comment]
@apottere I'm busy but I will try to find the time to fix this issue.

**Comment by mpires** [comment]
Hey guys I went ahead and fixed this, the pull request is #80
@apottere would you be able to merge it? Additionally, do you have any plans for the next release?

**Comment by mpires** [comment]
Hang on, I noticed that while it fixes the GraphQLNonNull situation, it can be extended to do the same for the issue @joel-costigliola raised. I'm adding that to the PR

**Comment by apottere** [comment]
I haven't released this library yet, but I can probably set it up today.  Might take a day or two to get it synced.

**Comment by mpires** [comment]
Meanwhile I think this change in the PR fixes the issue raised by @joel-costigliola

**Comment by tdraier** [comment]
This issue is also more globally fixed in #94
